cmd/kube-controller-manager/app/certificates.go

```diff
@@ -222,5 +228,50 @@ func startRootCACertificatePublisherController(ctx context.Context, controllerCo
 		return nil, true, fmt.Errorf("error creating root CA certificate publisher: %v", err)
 	}
 	go sac.Run(ctx, 1)
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ClusterTrustBundle) {
+		return nil, true, nil
+	}
+
+	apiserverSignerClient := controllerContext.ClientBuilder.ClientOrDie("kube-apiserver-serving-clustertrustbundle-publisher")
+	ctbAvailable, err := clusterTrustBundlesAvailable(apiserverSignerClient)
+	if err != nil {
+		return nil, true, fmt.Errorf("discovery failed for ClusterTrustBundle: %w", err)
+	}
+
+	if !ctbAvailable {
+		return nil, true, nil
+	}
+
+	servingSigners, err := dynamiccertificates.NewStaticCAContent("kube-apiserver-serving", rootCA)
+	if err != nil {
+		return nil, true, fmt.Errorf("failed to create a static CA content provider for the kube-apiserver-serving signer: %w", err)
+	}
+
+	ctbPublisher, err := ctbpublisher.NewClusterTrustBundlePublisher(
+		"kubernetes.io/kube-apiserver-serving",
+		servingSigners,
+		apiserverSignerClient,
+	)
+	if err != nil {
+		return nil, true, fmt.Errorf("error creating kube-apiserver-serving signer certificates publisher: %w", err)
+	}
+
+	go ctbPublisher.Run(ctx)
```

Comment:
Also I think the argument could be made that this should be in its own controller init function instead of `startRootCACertificatePublisherController` because that would allow you to disable the old one to save resources.